server/service/src/sync/synchroniser.rs

```diff
@@ -151,22 +151,30 @@ pub fn integrate_and_translate_sync_buffer(
     TranslationAndIntegrationResults,
     TranslationAndIntegrationResults,
 )> {
-    let sync_buffer = SyncBuffer::new(connection);
-    let translation_and_integration = TranslationAndIntegration::new(connection, &sync_buffer);
-
-    // Translate and integrate upserts (ordered by referencial database constraints)
-    let upsert_sync_buffer_records =
-        sync_buffer.get_ordered_sync_buffer_records(SyncBufferAction::Upsert)?;
-    let upsert_integration_result = translation_and_integration
-        .translate_and_integrate_sync_records(upsert_sync_buffer_records)?;
-
-    // Translate and integrate delete (ordered by referencial database constraints, in reverse)
-    let delete_sync_buffer_records =
-        sync_buffer.get_ordered_sync_buffer_records(SyncBufferAction::Delete)?;
-    let delete_integration_result = translation_and_integration
-        .translate_and_integrate_sync_records(delete_sync_buffer_records)?;
-
-    Ok((upsert_integration_result, delete_integration_result))
+    // Integration is done inside of transaction, to make sure all records are available at the same time
+    // and maintain logical data integrity
```

Comment:
Don't we assume there is only one sync process running at a time? do we need a transaction then?

Reply:
But it could be stopped half way and only half of the invoice would be available (someone stops the server say).
It sounds like we are still open to some of the translation and integration errors being a stopper for sync as a whole (but for now just skipping them all), at which point records in transaction are in 'undefined' state, so have to roll back ('undefined' being we can't guarantee logical data integrity)